src/app/components/FeeSelectField/index.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Form, Alert, Slider } from 'antd';
+import { AppState } from 'store/reducers';
+import Loader from 'components/Loader';
+import { SliderValue } from 'antd/lib/slider';
+import { CHAIN_TYPE } from 'utils/constants';
+import { getOnChainFeeEstimates } from 'modules/payment/actions';
+import { getNodeChain } from 'modules/node/selectors';
+import './style.less';
+
+interface StateProps {
+  onChainFees: AppState['payment']['onChainFees'];
+  feesError: AppState['payment']['feesError'];
+  isFetchingFees: AppState['payment']['isFetchingFees'];
+  chain: ReturnType<typeof getNodeChain>;
+}
+
+interface DispatchProps {
+  getOnChainFeeEstimates: typeof getOnChainFeeEstimates;
+}
+
+interface OwnProps {
+  showFeeMsg?: boolean;
+  onChange(fee: number): void;
+}
+
+type Props = StateProps & DispatchProps & OwnProps;
+
+interface State {
+  fee: number | undefined,
+}
+
+class FeeSelectField extends Component<Props, State> {
+  state: State = {
+    fee: 0,
+  }
+
+  componentDidMount() {
+    this.props.getOnChainFeeEstimates();
+  }
+
+  componentWillReceiveProps(nextProps: Props) {	
```

Comment:
FYI `componentWillReceiveProps` will be deprecated in React 17 and generally a good upgrade path for this kind of function is to use `componentDidUpdate` instead and compare against `prevProps` instead of `nextProps`

https://reactjs.org/docs/react-component.html#updating

More of a long term consideration but noticed it because I write a ton of React and have gotten used to it by now